flytepropeller/pkg/compiler/transformers/k8s/workflow.go

```diff
@@ -231,7 +249,19 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 		errs.Collect(errors.NewWorkflowBuildError(err))
 	}
 
-	obj.ObjectMeta.Name = name
+	hashedIdentifier := hashIdentifier(core.Identifier{
+		Project: project,
+		Domain:  domain,
+		Name:    name,
+	})
+	rand.Seed(int64(hashedIdentifier))
+
+	if workflowCRNameHashLength := config.GetConfig().WorkflowCRNameHashLength; workflowCRNameHashLength > 0 {
+		obj.ObjectMeta.Name = rand.String(workflowCRNameHashLength)
```

Comment:
Updated https://github.com/flyteorg/flyte/pull/5480#discussion_r1688491088

Now, for a setting like `WorkflowCRNameHashLength: 32`, the CR name will look like `mywfname-f5ptmmd66pktp59nphfdjkqr9szp8fv5` or `mywfname-wnw9zlg22z5wmdjggdcrnzdrpw59m4bc`. Note that the suffix is deterministic, and uses a hash of the name+project+domain.

I opted for this approach rather than `mywfname-project`, this is because of the character limit of the CR name and how this affects executions that have a long execution name and a long project name, where improperly handled truncation will lead to undesired behavior such as duplicates.